src/posix/platform/radio.cpp

```diff
@@ -205,6 +209,56 @@ void Radio::Init(void)
 
 void *Radio::GetSpinelInstance(void) { return &sRadioSpinel; }
 
+void Radio::GetIidListFromRadioUrl(spinel_iid_t *aIidList, const uint8_t aLength)
+{
+    // Validate the arguments
+    VerifyOrDie(aIidList != nullptr, OT_EXIT_INVALID_ARGUMENTS);
+    VerifyOrDie(aLength != 0 && aLength <= Spinel::kSpinelHeaderMaxNumIID, OT_EXIT_INVALID_ARGUMENTS);
+
+    // Initialize the aIidList with invalid iid values.
+    memset(aIidList, SPINEL_HEADER_INVALID_IID, aLength * sizeof(spinel_iid_t));
+
+    // Get IID strings
+    const char *iidString     = (mRadioUrl.GetValue("iid"));
+    const char *iidListString = (mRadioUrl.GetValue("iid-list"));
+
+#if OPENTHREAD_CONFIG_MULTIPAN_RCP_ENABLE
+    // First entry to the aIidList must be the IID of the host application.
+    VerifyOrDie(iidString != nullptr, OT_EXIT_INVALID_ARGUMENTS);
+    aIidList[0] = static_cast<spinel_iid_t>(atoi(iidString));
+
+    if (iidListString != nullptr)
+    {
+        // Convert string to an array of integers.
+        // Integer i is for traverse the iidListString.
+        // Integer j is for aIidList array offset location.
+        // First entry of aIidList is for host application iid hence j start from 1.
+        for (uint8_t i = 0, j = 1; iidListString[i] != '\0' && j < Spinel::kSpinelHeaderMaxNumIID; i++)
+        {
+            if (iidListString[i] == ',')
+            {
+                j++;
+                continue;
+            }
+            else if (iidListString[i] < '0' || iidListString[i] > '9')
+            {
+                DieNow(OT_EXIT_INVALID_ARGUMENTS);
+            }
```

Comment:
Can remove `else` after `continue` (I think we can some code-checkers, maybe `clang-tidy` that give warning for such a pattern)

```suggestion
            if (iidListString[i] == ',')
            {
                j++;
                continue;
            }
            
            if (iidListString[i] < '0' || iidListString[i] > '9')
            {
                DieNow(OT_EXIT_INVALID_ARGUMENTS);
            }
```

Reply:
I was not aware of that, thanks for info. Made suggested changes.